File: sudoku-common/common-core/src/main/java/com/zumin/sudoku/common/core/exception/FileException.java
package com.zumin.sudoku.common.core.exception;

import com.zumin.sudoku.common.core.enums.CommonStatusCode;

/**
 * 文件异常类
 */
public class FileException extends BaseException {

  private static final long serialVersionUID = -658002836273149602L;

  public FileException() {
    super(CommonStatusCode.FILE_ERROR);
  }

  public FileException(CommonStatusCode commonStatusCode) {
    super(commonStatusCode);
  }

  public FileException(Throwable cause) {
    super(CommonStatusCode.FILE_ERROR, cause);
  }

}

File: sudoku-common/common-core/src/main/java/com/zumin/sudoku/common/core/exception/FileException.kt
package com.zumin.sudoku.common.core.exception

import com.zumin.sudoku.common.core.exception.BaseException
import com.zumin.sudoku.common.core.code.CommonStatusCode

/**
 * 文件异常类
 */
class FileException : BaseException {
  constructor() : super(CommonStatusCode.FILE_ERROR)
  constructor(commonStatusCode: CommonStatusCode) : super(commonStatusCode)
  constructor(cause: Throwable) : super(CommonStatusCode.FILE_ERROR, cause)

  companion object {
    private const val serialVersionUID = -658002836273149602L
  }
}